manifests/config.pp

```diff
@@ -255,28 +213,14 @@
         notify  => Class['Rabbitmq::Service'],
       }
     }
-    'Archlinux': {
-      file { '/etc/systemd/system/rabbitmq.service.d':
-        ensure                  => directory,
-        owner                   => '0',
-        group                   => '0',
-        mode                    => '0755',
-        selinux_ignore_defaults => true,
-      }
-      -> file { '/etc/systemd/system/rabbitmq.service.d/limits.conf':
-        content => template('rabbitmq/rabbitmq-server.service.d/limits.conf'),
-        owner   => '0',
-        group   => '0',
-        mode    => '0644',
-        notify  => Exec['rabbitmq-systemd-reload'],
-      }
-      exec { 'rabbitmq-systemd-reload':
-        command     => '/bin/systemctl daemon-reload',
-        notify      => Class['Rabbitmq::Service'],
-        refreshonly => true,
-      }
-    }
-    default: {
+    default: { }
+  }
+
+  if $facts['systemd'] { # systemd fact provided by systemd module
+    systemd::service_limits { 'rabbitmq.service':
```

Comment:
Shouldn't this use $::rabbitmq::service_name?

On Debian Stretch, the Service is called "rabbitmq-server.service" - but with "rabbitmq.service", the override does not get applied.